src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/settings/ProjectDetailsAjaxController.java

```diff
@@ -0,0 +1,186 @@
+package ca.corefacility.bioinformatics.irida.ria.web.projects.settings;
+
+import java.util.Locale;
+
+import javax.validation.ConstraintViolationException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import ca.corefacility.bioinformatics.irida.exceptions.EntityNotFoundException;
+import ca.corefacility.bioinformatics.irida.model.project.Project;
+import ca.corefacility.bioinformatics.irida.model.workflow.submission.AnalysisSubmission;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.ajax.AjaxErrorResponse;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.ajax.AjaxResponse;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.dto.ajax.AjaxSuccessResponse;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.projects.settings.dto.Coverage;
+import ca.corefacility.bioinformatics.irida.ria.web.ajax.projects.settings.exceptions.UpdateException;
+import ca.corefacility.bioinformatics.irida.ria.web.projects.dto.ProjectDetailsResponse;
+import ca.corefacility.bioinformatics.irida.ria.web.projects.settings.dto.UpdateProjectAttributeRequest;
+import ca.corefacility.bioinformatics.irida.ria.web.services.UIMetadataService;
+import ca.corefacility.bioinformatics.irida.ria.web.services.UIProjectsService;
+import ca.corefacility.bioinformatics.irida.service.ProjectService;
+
+import com.google.common.base.Strings;
+
+/**
+ * Handle asynchronous requests for the UI project details page.
+ */
+@RestController
+@RequestMapping("/ajax/project/details")
+public class ProjectDetailsAjaxController {
+	private final ProjectService projectService;
+	private final UIProjectsService service;
+	private final UIMetadataService metadataService;
+	private final MessageSource messageSource;
+
+	@Autowired
+	public ProjectDetailsAjaxController(ProjectService projectService, UIProjectsService service,
+			UIMetadataService metadataService, MessageSource messageSource) {
+		this.projectService = projectService;
+		this.service = service;
+		this.metadataService = metadataService;
+		this.messageSource = messageSource;
+	}
+
+	/**
+	 * Get general details about the project.
+	 *
+	 * @param projectId {@link Long} identifier for the project
+	 * @return {@link ResponseEntity} containing the project details
+	 */
+	@RequestMapping("")
+	public ResponseEntity<ProjectDetailsResponse> getProjectDetails(@RequestParam Long projectId) {
+		return ResponseEntity.ok(service.getProjectInfo(projectId));
```

Comment:
Totally missed this before but we should probably have this in a try/catch and throw an error if there was issue getting the project details